src/mol-io/common/string-like.ts

```diff
@@ -0,0 +1,287 @@
+            }
+            out.push(this._chunks[iLastChunk].substring(0, indexInChunkTo));
+            return out.join('');
+        }
+    }
+    private readonly _tmpArray: string[] = [];
+    private _getTmpArray() {
+        while (this._tmpArray.length) this._tmpArray.pop(); // this seems to be faster than `this._tmpArray.length = 0` for short arrays
+        return this._tmpArray;
+    }
+
+    indexOf(searchString: string, position: number = 0): number {
+        if (searchString.length > this.STRING_CHUNK_SIZE) {
+            throw new Error('NotImplementedError: indexOf is only implemented for searchString shorter than STRING_CHUNK_SIZE');
+            // In real use-cases STRING_CHUNK_SIZE is big and it doesn't make sense to search for such long substrings.
+        }
+
+        if (position < 0) position = 0;
+        const iFirstChunk = this._getChunkIndex(position);
+
+        for (let iChunk = iFirstChunk; iChunk < this._chunks.length; iChunk++) {
+            const chunk = this._chunks[iChunk];
+            const positionInChunk = iChunk === iFirstChunk ? this._getIndexInChunk(position) : 0;
+
+            // Try to find the whole substring in this chunk
+            const found = chunk.indexOf(searchString, positionInChunk);
+            if (found >= 0) return iChunk * this.STRING_CHUNK_SIZE + found;
+
+            // Try to find the substring overflowing to the next chunk (assumes searchString.length <= STRING_CHUNK_SIZE)
+            if (iChunk !== this._chunks.length - 1) {
+                const start = Math.max(this.STRING_CHUNK_SIZE - searchString.length + 1, positionInChunk);
+                const aroundBoundary = chunk.substring(start, undefined) + this._chunks[iChunk + 1].substring(0, searchString.length - 1);
+                const found = aroundBoundary.indexOf(searchString);
+                if (found >= 0) return iChunk * this.STRING_CHUNK_SIZE + start + found;
+            }
+        }
+        return -1;
+    }
+
+    includes(searchString: string, position: number = 0): boolean {
+        return this.indexOf(searchString, position) >= 0;
+    }
+
+    startsWith(searchString: string, position: number = 0): boolean {
+        if (searchString.length > this.STRING_CHUNK_SIZE) {
+            throw new Error('NotImplementedError: startsWith is only implemented for searchString shorter than STRING_CHUNK_SIZE');
+            // In real use-cases STRING_CHUNK_SIZE is big and it doesn't make sense to search for such long substrings.
+        }
+        return this.substring(position, position + searchString.length) === searchString;
+    }
+
+    toString(): string {
+        try {
+            return this._chunks.join('');
+        } catch (err) {
+            throw new Error(`Failed to convert StringLike object into string. This might be because the length ${this.length} exceeds maximum allowed string length ${MAX_STRING_LENGTH}. (${err})`);
+        }
+    }
+
+    _repr(): string {
```

Comment:
what is this for?

Reply:
This was used in debugging. Yes, i can remove it.

Reply:
Removed.